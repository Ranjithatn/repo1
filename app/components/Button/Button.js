import React, { Fragment } from "react";
import Icon from "../Icon/Icon";

const Button = ({
  leftIcon,
  rightIcon,
  text,
  onClick,
  className,
  disabled = false,
  id,
  type = "button",
  style={},
}) => {
  return (
    <button
      id={id}
      type={type}
      className={"button" + (className !== undefined ? " " + className : "")}
      disabled={disabled}
      onClick={onClick}
      style={ style }
    >
      {leftIcon && <Icon icon={leftIcon} />}
      { text &&<span>{text}</span> }
      {rightIcon && <Icon icon={rightIcon} />}
    </button>
  );
};

export default Button;
