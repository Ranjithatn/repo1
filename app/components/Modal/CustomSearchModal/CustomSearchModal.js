import React, { Fragment } from "react";
import { connect } from "react-redux";

import Modal from "../Modal";
import Title from "../../Title/Title";
import Button from "../../Button/Button";
import Switch from "../../Switch/Switch";
import { requestHideModal } from "../../../actions/modal";
import { requestInputFieldChanged } from "../../../actions/global";
import { requestNewAction, requestActionStart } from "../../../actions/jobs";
import { selectedJobSelector, NewSearchActionSelector,ActionStartSelector } from "selectors/jobs";
import { usernameSelector } from "selectors/auth";
import { customSearchDataSelector } from "../../../selectors/customSearch";
import Permissions from "../../Permissions";
import "./CustomSearchModal.scss";


export const CustomSearchModal = ({
  requestHideModal,
  formatMessage,
  selectedAction,
  requestInputFieldChanged,
  requestNewAction,
  newAction,
  selectedJob,
  username,
  all_actions,
  customSearchData,
  requestActionStart,
  actionStart
}) => {



  return (
    <Modal
      title={formatMessage({ id: "submitPrints" })}
      className="newAction-modal"
      requestHideModal={requestHideModal}
      content={
        <Fragment>
          <Title
            is="5"
            text={formatMessage({ id: "newAction.searchOptions" })}
          />
          <Permissions version={2} type="menu.criminal.search">
          <Switch
            id="searchCriminalDB"
            onChange={e => requestInputFieldChanged(e, "newSearchAction")}
            value="Search Criminal"
            text={formatMessage({ id: "searchCriminalDB" })}
            className="is-primary is-rounded"
            checked={
              // newAction.includes("Search Civil") ||
              newAction.includes("Search Criminal")
                ? true
                : false
            }
            // disabled={
            //   newAction.includes("Search Civil")
            //     ? true
            //     : false
            // }
            disabled={
              customSearchData.fileNo === "" || !customSearchData.fileNo
            }
          />
       
         </Permissions>
          <Permissions version={2} type="menu.latent.customsearch">
            <Switch
              id="searchLatentDB"
              onChange={e =>
                requestInputFieldChanged(e, "newCustomSearchAction")
              }
              value="Search Latent"
              text={formatMessage({ id: "searchLatentDB" })}
              className="is-primary is-rounded"
              disabled={
                customSearchData.latentID === "" || !customSearchData.latentID
              }
              // checked={newAction === "Search Civil"}
            />
          </Permissions>
          <Permissions version={2} type="menu.civil.search">
            <Switch
              id="searchCivilDB"
              onChange={e =>
                requestInputFieldChanged(e, "newCustomSearchAction")
              }
              value="Search Civil"
              text={formatMessage({ id: "searchCivilDB" })}
              className="is-primary is-rounded"
              disabled={
                customSearchData.SAMISID === "" || !customSearchData.SAMISID
              
              }
              // checked={newAction === "Search Civil"}
            />
          </Permissions>
        </Fragment>
      }
      buttons={
        <Fragment>
          <Button
            id="newAction-btn"
            className="is-primary"
            text={formatMessage({ id: "Submit" })}
            disabled={newAction.length <= 0 || actionStart}
            onClick={() => {
              requestActionStart(true);
              requestNewAction({
                newAction,
                job: selectedJob,
                username,
                refActionID: "",
                verifyAction: false,
                Note: "Action Created",
                typeOfAction: "Custom Search"
              });
              // requestHideModal();
            }}
          />
        </Fragment>
      }
    />
  );
};

const mapState = state => ({
  all_actions: state.jobs.newSearchAction,
  newAction: NewSearchActionSelector(state),
  selectedJob: selectedJobSelector(state),
  username: usernameSelector(state),
  actionStart:ActionStartSelector(state),
  customSearchData: customSearchDataSelector(state)
});
export default connect(
  mapState,
  {
    requestHideModal,
    requestInputFieldChanged,
    requestNewAction,
    requestActionStart
  }
)(CustomSearchModal);
