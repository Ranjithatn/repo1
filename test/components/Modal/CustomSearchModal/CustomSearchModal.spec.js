import React from "react";
import toJson from "enzyme-to-json";
import { CustomSearchModal } from "../../../../app/components/Modal/CustomSearchModal/CustomSearchModal";
import Button from "../../../../app/components/Button/Button";
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

let component, tree, instance;
let tenprintToggle,searchCivilDBToggle,newActionButton;

let props = {
  requestHideModal: jest.fn(),
  requestInputFieldChanged: jest.fn(),
  formatMessage: jest.fn(),
  requestNewAction: jest.fn(),
  newAction:"Search Criminal",
  customSearchData:{
      SAMISID:"298"
    }
}

beforeEach(() => {
  component = mount(
    <CustomSearchModal {...props} />
  );
  tenprintToggle = component.find("input#searchCriminalDB");
  searchCivilDBToggle = component.find("input#searchCivilDB");
  newActionButton = component.find("button#newAction-btn");
  tree = toJson(component);
  instance = component.instance();

});

describe("<CustomSearchModal />", () => {
    xit("should render without crashing", () => {
      expect(tree).toMatchSnapshot();
    });
    xit("should call requestInputFieldChanged when tenprint is toggled", () => {
        let event: {
          id: "searchCriminalDB"
        };
        tenprintToggle.simulate("change", event);
        expect(props.requestInputFieldChanged).toHaveBeenCalled();
      });
    xit("should call requestInputFieldChanged when tenprint is toggled", () => {
        let event: {
          id: "searchCivilDB"
        };
        searchCivilDBToggle.simulate("change", event);
        expect(props.requestInputFieldChanged).toHaveBeenCalled();
      });

      xit('should call requestCreateJob when save job btn is clicked', () => {
        const evt = {
          target: { id: "clicked" }
        }
        newActionButton.simulate('click', evt);
        expect(props.requestNewAction).toHaveBeenCalled();
      })
})