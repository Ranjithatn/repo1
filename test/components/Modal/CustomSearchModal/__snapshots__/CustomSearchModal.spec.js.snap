// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomSearchModal /> should render without crashing 1`] = `undefined`;
